lib/engine/game/g_18_scan/game.rb

```diff
@@ -259,18 +287,187 @@ def stock_round
           ])
         end
 
-        def train_limit(entity)
-          super + Array(abilities(entity, :train_limit)).sum(&:increase)
+        def operating_round(round_num)
+          Round::Operating.new(self, [
+            Engine::Step::Bankrupt,
+            G18Scan::Step::Track,
+            # G18Scan::Step::DestinationToken
+            # G18Scan::Step::DestinationRun
+            Engine::Step::Token,
+            # G18Scan::Step::BonusToken
+            Engine::Step::Route,
+            G18Scan::Step::Dividend,
+            Engine::Step::DiscardTrain,
+            G18Scan::Step::BuyTrain,
+          ], round_num: round_num)
+        end
+
+        def ipo_name
+          'Treasury'
         end
 
         def sj
           @sj ||= corporation_by_id('SJ')
         end
 
+        def mine
+          @mine ||= company_by_id('Mine')
+        end
+
+        def mine_included(route)
+          route.corporation.assigned?(mine.id) && route.hexes.find { |h| h.id == 'A20' }
```

Comment:
don't use find if you're not using the return value, use any? same with line 326